src/book/book.service.ts

```diff
@@ -1,28 +1,113 @@
-import { Injectable, Logger } from "@nestjs/common";
+import { ConflictException, Injectable, Logger, NotFoundException } from "@nestjs/common";
+
+import { ProcessFile, formatBookName } from "@/utils";
 import { CreateBookDto } from "./dto/create-book.dto";
 import { UpdateBookDto } from "./dto/update-book.dto";
+import { UploadBookDto } from "./dto/upload-book.dto";
+import { BookRepository } from "./models/book.repository";
 
 @Injectable()
 export class BookService {
-  create(createBookDto: CreateBookDto) {
-    Logger.log(createBookDto);
-    return "This action adds a new book";
+  constructor(private readonly bookRepository: BookRepository) {}
+
+  private async processUploadedFiles(f: UploadBookDto, bookName: string) {
+    const coverImage: Buffer | undefined = f.cover?.[0].buffer;
+    const pdfFile: Buffer | undefined = f.file?.[0].buffer;
+    const formattedBookName: string = formatBookName(bookName);
+
+    const pImg: ProcessFile = new ProcessFile({
+      fileBuffer: coverImage,
+      fileType: "cover",
+      subDir: formattedBookName,
+    });
+    const pPdf: ProcessFile = new ProcessFile({
+      fileBuffer: pdfFile,
+      fileType: "file",
+      customExt: "pdf",
+      subDir: formattedBookName,
+    });
+
+    const coverPath: string = await pImg.resizeAndSave({ width: 432, height: 574 });
+    const bookPath: string = await pPdf.saveFile();
```

Comment:
Any particular reason behind this values?

Reply:
By this value, you mean the width and hight value ? There is no reason why it is 432x574, I felt this is small enough to not cause a lot of issues, but not too small so it is still usable without scaling the cover image on the frontend.

If by the value you mean `coverPath` and `bookPath`, they are the paths on where the `PDF` and the cover `img` are saved on the server